python/cudf/cudf/utils/dtypes.py

```diff
@@ -572,6 +572,24 @@ def find_common_type(dtypes):
             )
         else:
             return cudf.dtype("O")
+    if any(cudf.api.types.is_list_dtype(dtype) for dtype in dtypes):
+        if len(dtypes) == 1:
+            return dtypes.get(0)
+        else:
+            # TODO: As list dtypes allow casting
+            # to identical types, improve this logic of returning a
+            # common dtype, for example:
+            # ListDtype(int64) & ListDtype(int32) common
+            # dtype could be ListDtype(int64).
+            return cudf.dtype("O")
```

Comment:
Is this really what we want to do here? With pandas we get a true object type, but with cudf this will result in a string. The new concat test seems pretty weird in that respect. We take an int column and a list column and end up with the string representations. @shwina WDYT?

Reply:
I agree, I'm inclined to prefer an error message rather than a change to a string as you say in your review as well.

Reply:
Thanks @vyasr & @wence-, I agree it makes sense to raise an error rather than casting to string types.